client/src/app/api/api/api.service.ts

```diff
@@ -213,6 +225,39 @@ export class ApiService {
     );
   }
 
+  private postLabel(endpoint: string, idInstance: string, labelName: string, observe: any = 'body', reportProgress: boolean = false): any {
+    if (idInstance === null || idInstance === undefined && labelName === null || labelName === undefined) {
+      throw new Error('Required parameter instanceId and Instance Name was null or undefined when calling getInstanceNumber.');
```

Comment:
you should alter the error message, we are not calling getInstanceNumber here and neither are we checking the instance name parameter...